app/controllers/clinics_controller.rb

```diff
@@ -23,14 +23,22 @@ def new
     @clinic = Clinic.new
   end
 
-  # def update
-  #   clinic = Clinic.find params[:id]
-  #   if clinic.save
-  #     head :ok
-  #   else
-  #     head :bad_request
-  #   end
-  # end
+  def edit
+  end
+
+  def update
+    if @clinic.update_attributes clinic_params
+      flash[:notice] = 'Successfully updated clinic details'
+      redirect_to clinics_path
+    else
+      flash[:notice] = 'Error saving clinic details'
```

Comment:
Could this please `flash[:alert]` instead of `flash[:notice]`? That will make it appear styled red to grab attention on failures.